lib/coverband/reporters/base.rb

```diff
@@ -2,25 +2,26 @@
 
 module Coverband
   module Reporters
+    ###
+    # This is the base clase for report generation
+    # it helps with filtering, normalization, etc for final reprort generation
+    ###
     class Base
-      def self.report(store, options = {})
-        roots = get_roots
-        additional_coverage_data = options.fetch(:additional_scov_data) { [] }
+      def self.report(store, _options = {})
+        scov_style_report = get_current_scov_data_imp(store, root_paths)
 
         if Coverband.configuration.verbose
-          Coverband.configuration.logger.info "fixing root: #{roots.join(', ')}"
-          Coverband.configuration.logger.debug "additional data:\n #{additional_coverage_data}"
-        end
-
-        scov_style_report = report_scov_with_additional_data(store, additional_coverage_data, roots)
-
-        if Coverband.configuration.verbose
-          Coverband.configuration.logger.debug "report:\n #{scov_style_report.inspect}"
+          msg = "report:\n #{scov_style_report.inspect}"
+          Coverband.configuration.logger.debug msg
         end
         scov_style_report
       end
 
-      def self.get_roots
+      # protected
+      # below are all not public API
+      # but as they are class methods protected doesn't really do anything
```

Comment:
Can we use private_class_method declaration?

https://apidock.com/ruby/Module/private_class_method

Reply:
interesting I will look into this

Reply:
fixed